### DISCUSSION TITLE
Are these things possible with Transmission?

### DISCUSSION THREAD

**Discussion by astronomia123** [transmission/transmission#5366] [Q&A]
PS : I've never used transmission from the command line. 

Hi, I'm a Laravel (PHP) developer, and I want to create an application that runs on a remote server for downloading torrents with a certain degree of automation and recording/logging.

What I'm trying to do is this
Add Magnet links to DB -> Retrieve them and Download Torrents -> Upload them to a remote server -> delete the downloaded files
And keep a record of the entire process in a database

Before I dwell in the docs and go down a Google search rabbit hole (I've already spent 2 hours doing this). I just want to ask if its even possible.

So, My questions are...can I do the following from within PHP/Laravel?

1. Add torrents using magnet URIs retrieved from a database. (Hourly CRON job)
  - Limit max 10 active torrents. Or maybe only add 10 torrents
2. Attach a unique ID to each torrent and save it to the database.
3. Identifying that a torrent has finished downloading. (An event fires a function in Laravel, or hourly CRON job to identify)
  - Retrieve the unique ID in step 2 for the downloaded torrents. in order to update their status in the database.
  - Trigger upload to remote servers. (These remote servers have APIs that support uploading)
  - Once upload finishes, delete the downloaded files

I'd really appreciate it if someone could tell me before had if what I'm trying to do is possible, before I spend hours researching on it.
Also, If you could point me in the right direction, that would be great… :) 

I'm sorry if these questions are basic... but as I said previously... I'm completely new to transmission CLI, I've just used the app on Windows

**Comment by robd003** [comment]
Yes, all of this is possible. Take a look at transmission-remote

You could also write a library to interact with the transmission-daemon RPC system. Here's some details on the RPC implementation: https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md

**Reply by astronomia123** [reply]
Hey... Sry For the late reply... Thank you for the response and **that Link**.. I dived into it and forgot to respond....😅😅😅

I was able to figure out most of my requirements from the RPC documents (I also found a library for Laravel... so that would work the best)

What I couldn't find was the ability to give a torrent a unique ID. My fallback mechanism would be to use the torrent hash, but if there's a way to give a unique ID... That would make things a lot more easier... Any Idea?

**Reply by ckerr** [reply]
There's not a dedicated assignable ID featture, but you can assign labels to each torrent. You could generate a new, unique label per-torrent and use that.

**Comment by ksurl** [comment]
sounds like all this can be managed with flexget
